### PR TITLE
Malformed sticky_posts option breaks sticky posts.

### PR DESCRIPTION
**Pull request by peterwilsoncc**
https://core.trac.wordpress.org/ticket/55176

### CODE REVIEW

**Comment by peterwilsoncc** [comment]
@costdev Would you be able to take a look at this PR, for some reason I can not assign you as a reviewer.

File: tests/phpunit/tests/post.php [hunk 1]
```diff
@@ -1685,6 +1685,57 @@ public function data_stick_post_does_not_duplicate_post_ids() {
...
+
+	/**
+	 * Ensures sticking works with an unexpected option value.
+	 *
+	 * @ticket 52007
+	 * @ticket 55176
+	 * @covers ::stick_post
+	 * @dataProvider data_stick_post_with_unexpected_sticky_posts_option
+	 *
+	 * @param mixed $stick Value to pass to stick_post().
```
Comment (costdev) [inline tests/phpunit/tests/post.php]:
```suggestion
	 * @param mixed $starting_option Value to pass to stick_post().
```

File: tests/phpunit/tests/post.php [hunk 2]
```diff
@@ -1685,6 +1685,57 @@ public function data_stick_post_does_not_duplicate_post_ids() {
...
+	}
+
+	/**
+	 * Data provider for test_stick_post_after_delete_sticky_posts_option().
+	 *
+	 * @return array[] {
+	 *     Arguments passed to test.
+	 *
+	 *     @type mixed $stick Value to pass to stick_post().
+	 * }
```
Comment (costdev) [inline tests/phpunit/tests/post.php]: Data provider docblocks can be simply this:

```php
/**
 * Data provider.
 *
 * @return array
 */
```

This reduces maintainence churn for arguments, as well as when future test methods may also use the data provider.

File: tests/phpunit/tests/post.php [hunk 3]
```diff
@@ -1685,6 +1685,57 @@ public function data_stick_post_does_not_duplicate_post_ids() {
...
+	 */
+	public function data_stick_post_with_unexpected_sticky_posts_option() {
+		return array(
+			array( false ),
+			array( 'string' ),
+			array( 1 ),
+			array( null ),
+			array( true ),
+			array( new stdClass ),
```
Comment (costdev) [inline tests/phpunit/tests/post.php]:
```suggestion
			'false'     => array( false ),
			'a string'  => array( 'string' ),
			'1 int'     => array( 1 ),
			'null'      => array( null ),
			'true'      => array( true ),
			'an object' => array( new stdClass ),
```
While this looks redundant, see the summary of [this ticket](https://core.trac.wordpress.org/ticket/53119) (and the links).

**Review by costdev — CHANGES_REQUESTED** [review]
Just a few suggestions for consistency in the test suite.

**Comment by peterwilsoncc** [comment]
Committed in https://core.trac.wordpress.org/changeset/53238

**Pull request closed without merging**